input/full.go

```diff
@@ -35,18 +35,41 @@ func CollectFull(ctx context.Context, server *state.Server, connection *sql.DB,
 		return
 	}
 
+	// Perform one high frequency stats collection at the exact time of the full snapshot.
+	//
+	// The scheduler skips the otherwise scheduled execution when the full snapshot time happens,
+	// so we can run it inline here and pass its data along as part of this full snapshot.
+	server.HighFreqStateMutex.Lock()
+	newHighFreqState, err := CollectAndDiff1minStats(ctx, c, connection, ps.CollectedAt, server.HighFreqPrevState)
+	if err != nil {
+		logger.PrintError("Could not collect high frequency statistics for server: %s", err)
+		err = nil
+	} else {
+		// Move over previously collected high frequency stats to be submitted with this full snapshot
+		//
+		// We don't want to delay this longer in case query text or schema metadata collection takes time.
+		// Doing this later could mean that additional high frequency query stats are being collected and
+		// included in the submission, whose query texts were not yet picked up with this snapshot.
+		ts.StatementStats = newHighFreqState.UnidentifiedStatementStats
+		ts.PlanStats = newHighFreqState.UnidentifiedPlanStats
+		ts.ServerIoStats = newHighFreqState.QueuedServerIoStats
+		newHighFreqState.UnidentifiedStatementStats = make(state.HistoricStatementStatsMap)
+		newHighFreqState.UnidentifiedPlanStats = make(state.HistoricPlanStatsMap)
+		newHighFreqState.QueuedServerIoStats = make(state.HistoricPostgresServerIoStatsMap)
+		server.HighFreqPrevState = newHighFreqState
+	}
+	server.HighFreqStateMutex.Unlock()
 	bufferCacheReady := make(chan state.BufferCache)
 	go func() {
 		postgres.GetBufferCache(ctx, c, server, opts, bufferCacheReady)
 	}()
 
-	ps.LastStatementStatsAt = time.Now()
 	err = postgres.SetQueryTextStatementTimeout(ctx, connection, logger, server)
 	if err != nil {
 		logger.PrintError("Error setting query text timeout: %s", err)
 		return
 	}
-	ts.Statements, ts.StatementTexts, ps.StatementStats, err = postgres.GetStatements(ctx, c, connection, true)
+	ts.Statements, ts.StatementTexts, err = postgres.GetStatementTexts(ctx, c, connection)
```

Comment:
I wonder if this should be moved to a goroutine that's spun up at the start of the snapshot, similar to `GetBufferCache`, so other work can be done while waiting on Postgres to read the query text file. The one downside is that we'd potentially use more memory that way, since other collection tasks could be performed while `GetStatementTexts` has received the text and is doing work on it.

Reply:
Oh, but some customers might not want us to use another database connection. Instead we could have a single goroutine that calls both `GetBufferCache` and `GetStatementTexts`

Reply:
Another option would be to have three goroutines: buffer cache, statement texts, and everything else. A max of two goroutines would actually run in parallel, to reduce database connection usage. Though I'm not sure how to implement that in Go.

Reply:
I'm not concerned that the additional query stats collection will cause a slowdown, but we know query text collection is already a bottleneck. But yes it makes sense for that to be a separate PR.